tests/test_rds/test_rds.py

```diff
@@ -1076,6 +1076,48 @@ def test_copy_db_snapshots(delete_db_instance: bool):
     assert result["TagList"] == []
 
 
+@mock_aws
+def test_copy_db_snapshots_copytags_and_tags():
+    conn = boto3.client("rds", region_name=DEFAULT_REGION)
+    conn.create_db_instance(
+        DBInstanceIdentifier="db-primary-1",
+        Engine="postgres",
+        DBInstanceClass="db.m1.small",
+    )
+
+    original_tags = [{"Key": "original", "Value": "snapshot tags"}]
+    conn.create_db_snapshot(
+        DBInstanceIdentifier="db-primary-1",
+        DBSnapshotIdentifier="snapshot",
+        Tags=original_tags,
+    )
+
+    new_tags = [{"Key": "new", "Value": "tag"}]
+    kwargs_and_expected_results = [
```

Comment:
Nice! This is a very clean and clear way of showing the different behaviours.

Just one nitpick - can you use this list to parametrize the entire test? That is going to include a bit of overhead, but having separate tests for each use case makes debugging later on much easier (IMHO).

So the resulting test signature would look something like this:
```
@pytest.mark.parametrize("kwargs,expected_result", [..])
def test_copy_db_snapshots_copytags_and_tags(kwargs, expected_result):
    ...
```